### PR TITLE
Fixing filenames to prevent bad characters

### PR DESCRIPTION
**Pull request by jaimeburnap**
Thank you for contributing to **AlexaPi**.

**Issue this PR fixes (if applicable):** #309 

**Description:**

**Checklist:**
  - [ ] Update CHANGELOG.md
  - [ ] Update Wiki (new platform, installation, configuration changes etc)
  - [x] PR Location (usually `master` for hotfixes, `dev` for general development)

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -452,7 +453,7 @@ def process_response(response):
 			for directive in j['messageBody']['directives']:
 				if directive['namespace'] == 'SpeechSynthesizer':
 					if directive['name'] == 'speak':
-						player.play_speech(mrl_fix("file://" + tmp_path + directive['payload']['audioContent'].lstrip("cid:") + ".mp3"))
+						player.play_speech(mrl_fix("file://" + tmp_path + hashlib.md5(directive['payload']['audioContent'].lstrip("cid:")).hexdigest() + ".mp3"))
```
Comment (ksanghai) [inline src/main.py]: replace lstrip("cid:") with replace("cid:","",1) to avoid the case where lstrip strips cid:c or cid:i or cid:d

**Comment by renekliment** [comment]
Couple of things:

1. I would like a `logger.debug()` saying what filename was hashed to what.
2. Should this be fixed also in `play_playlist()`?
3. Do we still need the `mrl_fix()`? From a quick look I think it was there only to replace `#` which I guess is no longer required. I honestly don't remember exactly why I put it there.

Otherwise seems like a good solution :thumbsup:

**Comment by jaimeburnap** [comment]
@renekliment How can I test playlist changes? I'm under the impression that non-commercial devices don't work through this api? I think mrl_fix can be removed with this PR.

**Comment by renekliment** [comment]
@jaimeburnap Just ask for the news. This is what I get:

```
{
  "messageHeader": {},
  "messageBody": {
    "directives": [
      {
        "namespace": "Speaker",
        "name": "SetMute",
        "payload": {
          "mute": false
        }
      },
      {
        "namespace": "SpeechSynthesizer",
        "name": "speak",
        "payload": {
          "contentIdentifier": "amzn1.as-ct.v1.Domain:Application:DailyBriefing:TTS#ACRI#DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:Say:DAILYBRIEFING:DailyBriefingIntroduction",
          "audioContent": "cid:DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:Say:DAILYBRIEFING:DailyBriefingIntroduction_1986125804"
        }
      },
      {
        "namespace": "Speaker",
        "name": "SetMute",
        "payload": {
          "mute": false
        }
      },
      {
        "namespace": "AudioPlayer",
        "name": "play",
        "payload": {
          "audioItem": {
            "streams": [
              {
                "streamUrl": "cid:DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:ChannelIntroduction:0",
                "offsetInMilliseconds": 0,
                "progressReport": {
                  "progressReportDelayInMilliseconds": 0,
                  "progressReportIntervalInMilliseconds": 0
                },
                "progressReportRequired": false,
                "streamId": "amzn1.as-ct.v1.Domain:Application:DailyBriefing:TTS#ACRI#ssml#ACRI#DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:ChannelIntroduction:0"
              },
              {
                "streamUrl": "http://opml.radiotime.com/Tune.ashx?id=t116844467&sid=p14&formats=aac,mp3&partnerId=4JqugguZ&serial=AEMXM7MAWOTKJI75PHPZD4FTWCOQ",
                "offsetInMilliseconds": 0,
                "expiryTime": "1970-01-01T00:01:00+0000",
                "progressReport": {
                  "progressReportDelayInMilliseconds": 0,
                  "progressReportIntervalInMilliseconds": 0
                },
                "progressReportRequired": false,
                "streamId": "amzn1.as-ct.v1.Domain:Application:DailyBriefing:TTS#ACRI#url#ACRI#DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:ChannelItem:0:0"
              }
            ],
            "audioItemId": "amzn1.as-ct.v1.Domain:Application:DailyBriefing:TTS#ACRI#url#ACRI#DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:ChannelItem:0:0"
          },
          "navigationToken": "amzn1.as-ct.v1.Domain:Application:DailyBriefing:TTS#ACRI#url#ACRI#DailyBriefingPrompt.352710e7-04d7-42f5-8d83-3f3aff70bb5c:ChannelItem:0:0",
          "playBehavior": "REPLACE_PREVIOUS"
        }
      }
    ]
  }
}
```

**Comment by renekliment** [comment]
@jaimeburnap Any update?

**Comment by jaimeburnap** [comment]
@renekliment Got busy. Apologies. This should fix the three things you mentioned earlier.

**Comment by renekliment** [comment]
@jaimeburnap Thanks.

**Pull request merged**